### PR TITLE
Dsm no data pipe fix and test

### PR DESCRIPTION
**Pull request by GiCharkviani**
noData pipe adjustment to 0 case and also unit test for that pipe.

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {NoDataPipe} from './noData.pipe';
+import {expect} from '@angular/flex-layout/_private-utils/testing';
+
+describe('noData Pipe', () => {
+  const pipe = new NoDataPipe();
+
+  const nonBrakingSpace = (): string => {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.spec.ts]: a typo, should be `nonBreakingSpace`

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.spec.ts]: Thanks 🙏

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {Pipe, PipeTransform} from '@angular/core';
 export class NoDataPipe implements PipeTransform {
   transform(value: any): string {
     const tempElement = document.createElement('div');
-    tempElement.innerHTML = value || '&nbsp;';
+    tempElement.innerHTML = (typeof value === 'number' ? String(value) : value) || '&nbsp;';
     return tempElement.innerText;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: Seems like you can use just `String.fromCharCode(160)` for the non-breaking space 
(see [link](https://stackoverflow.com/questions/5237989/how-is-a-non-breaking-space-represented-in-a-javascript-string)).

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: Better, Thank you

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {Pipe, PipeTransform} from '@angular/core';
 export class NoDataPipe implements PipeTransform {
   transform(value: any): string {
     const tempElement = document.createElement('div');
-    tempElement.innerHTML = value || '&nbsp;';
+    tempElement.innerHTML = (typeof value === 'number' ? String(value) : value) || '&nbsp;';
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: What type the value is supposed to be?
If it is only **number** / **string** / **boolean** maybe using `value.toString()` is just enough.

Comment (GiCharkviani) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: it can be anything

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: including null and undefined

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts (hunk 2 above)
Comment (alextanlan) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: If the value is just **number** / **string** / **boolean**, 
may be, it is enough to have smth like:

```
transform(value: number | string | boolean): string {
    const nonBreakingSpace = String.fromCharCode(160);
    return (value || value === 0) ? String(value) : nonBreakingSpace;
}
```

File: ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts [hunk 4]
```diff
@@ -6,8 +6,6 @@ import {Pipe, PipeTransform} from '@angular/core';
 
 export class NoDataPipe implements PipeTransform {
   transform(value: any): string {
-    const tempElement = document.createElement('div');
-    tempElement.innerHTML = value || '&nbsp;';
-    return tempElement.innerText;
+    return (value === 0 ? value.toString() : value) || String.fromCharCode(160);
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: Would you add ` const nonBreakingSpace = String.fromCharCode(160);`?
It is not obvious from the first sight.

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/pipes/noData.pipe.ts]: Done!

**Review by alextanlan — APPROVED** [review]

**Pull request merged**